### ISSUE TITLE
Replace existing toast / remove all toasts

### ISSUE THREAD

**Issue by bradbarrow** [jossmac/react-toast-notifications#48]
Hi there, thanks for the library it's super simple and flexible :)

I was wondering if you had any thoughts about some sort of "replaceable" toast, or a toast limit of 1, if you will.

The idea being that a user might perform many repetitive actions that result in a toast. 
The toast is useful, but stacking lots of them on top of each other isn't necessary.
We'd like to `replaceToast()` in this case.

The way I've attempted to do this now, is simply to iterate over `toastStack` and `removeToast` for each of them before adding a new one.

However, I've found that due to transition timing, this can:

- take a long time
- cause the new (replacing) toast to bump the old toast out of the way as it replaces it (the old one is still busy leaving)

I tried using the callback of `removeToast` but it doesn't seem to wait until the transition is complete.

Any thoughts on this pattern? Other approaches I might try?

**Comment by bradbarrow** [comment]
Similarly, any thoughts on a persistent toast, that you could update whilst it's open?

**Comment by jossmac** [comment]
Hi @bradbarrow, thanks mate!

> The idea being that a user might perform many repetitive actions that result in a toast. The toast is useful, but stacking lots of them on top of each other isn't necessary.

Yeah, I've thought about that a bit. Rather than increasing the API surface area with `replaceToast()`, I think it might be tidier to allow consumers to pass an `id` along with the toast options. Internally it'd store only unique toasts by id, so you could call `addToast()` all you like, but while the id remains the same, only one would appear to the user.

> Similarly, any thoughts on a persistent toast, that you could update whilst it's open?

That could be handy, how about an `updateToast()` method? With the signature:

```js
type updateToast = (ToastId, NewOptionsObject) => void
```

**Comment by Jaikant** [comment]
+1 for this feature

**Comment by Jaikant** [comment]
I used this workaround:

```
const ToastComponent = React.forwardRef((props, ref) => {
  const [refresh, setRefresh] = useState(false);
  
  useImperativeHandle(ref, () => ({
    refresh: () => {
      setRefresh(!refresh);
    }
  }));

 return (<Text center> data.. ... {props.getProgress()}% </Text>)
})

export const Component = ({ progress }) => {
  const renderRef = useRef('render');
  const progressRef = useRef('progress');
  progressRef.current = progress;

  const getProgress = () => {  
    return progressRef.current 
  };
  const { addToast } = useToasts();

  useEffect(() => {
      addToast(<ToastComponent ref={renderRef} getProgress={getProgress}/>, { appearance: 'success' });
  
    if ( need to refresh) {
        renderRef.current.refresh();
    }
  },[..]);  
  return null;
};
```

**Comment by jossmac** [comment]
Changes described in #77 available in `2.4.0`.

**Comment by jossmac** [comment]
@bradbarrow @Jaikant @tafelito Changes described in #77 available in `2.4.0`.
